From fda4d81801e9d972c029fb4ed1162b960eb540b5 Mon Sep 17 00:00:00 2001
From: Alistair Francis <alistair.francis@wdc.com>
Date: Tue, 25 Aug 2020 11:15:27 -0700
Subject: [PATCH 052/267] io/lockf: Include bits/types.h before
 __OFF_T_MATCHES_OFF64_T check

It's possible that although __OFF_T_MATCHES_OFF64_T is defined the
included the relevent header file.  This results in a io/tst-lockf
failure for RV32 by calling the non 64-bit version of lockf.  This
patch fixes the failure by including bits/types.h.

Reviewed-by: Adhemerval Zanella <adhemerval.zanella@linaro.org>
---
 io/lockf.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/io/lockf.c b/io/lockf.c
index aa5c673c8d..8ea83e15f0 100644
--- a/io/lockf.c
+++ b/io/lockf.c
@@ -15,6 +15,8 @@
    License along with the GNU C Library; if not, see
    <https://www.gnu.org/licenses/>.  */
 
+#include <bits/types.h>
+
 #ifndef __OFF_T_MATCHES_OFF64_T
 
 #include <unistd.h>
-- 
2.29.0

